R/Utilities.R

```diff
@@ -64,15 +64,25 @@ autoclean <- function(.data) {
     toupper(.data$DetectionQuantitationLimitMeasure.MeasureUnitCode)
   # .data$BiologicalIntentName = toupper(.data$BiologicalIntentName)
   
-  # Remove duplicate rows
-  .data <- .data[!duplicated(.data), ]
+  # Remove duplicate rows - turned into a test because duplicated() takes a long
+  # time acting on all columns in a large dataset.
+  if(!length(unique(.data$ResultIdentifier))==dim(.data)[1]){
```

Comment:
Good thinking. This approach should work fine for removing exact duplicates. Let's add a test for this as well.

Reply:
I couldn't think of a good test for this so I went without. Let me know if you think of one.